compiler-core/src/erlang/tests/echo.rs

```diff
@@ -0,0 +1,124 @@
+macro_rules! assert_generic_echo {
```

Comment:
What's a generic echo? This is called "assert Erlang" in the other modules. Any reason to not to have this functionality in all codegen tests?

Reply:
It's the specific echo implementation that works on any generic type by doing runtime inspection, in the future the compiler could cut down on this code by only adding the implementation needed for a specific type if it can tell it at compile time:
```gleam
pub fn main() {
  echo 1
}
```
Could just become (it might be wrong but you get the idea):
```erl
main() ->
  echo(1, fun inspect@int/1).

echo(N, Inspect) ->
  String = Inspect(N),
  io:put_chars(standard_error, [String, $\n]),
  N.
  
inspect@int(N) ->
  erlang:integer_to_list(N).
```
So here I would want to just `assert_erl!` and look at the entirety of the code unlike in the generic case where I can just check that it contains that big chunk of code.